### PR TITLE
[APP-135] Promote social media

### PR DESCRIPTION
**Pull request by phillipsgoncalves**
**What does this PR do?**

This PR aims at implementing the social media promotion feature on 3 different views: MyAccount, Editorial, AppCoinsInfoView

**Database changed?**

 No

**Where should the reviewer start?**

- [ ] SocialMediaView.kt

**How should this be manually tested?**

Open the mentioned views and check the card. Also click on the buttons to see that it opened the correct page.
Also check for the analytics event that is being sent.

**What are the relevant tickets?**

  Tickets related to this pull-request: [APP-135](https://aptoide.atlassian.net/browse/APP-135)

**Questions:**

   Does this add new dependencies which need to be added? (Eg. new keys, etc.) 

**Code Review Checklist**

- [ ] Documentation on public interfaces
- [ ] Database changed? If yes - Migration?
- [ ] Remove comments & unused code & forgotten testing Logs
- [ ] Codestyle
- [ ] New Kotlin code has unit tests
- [ ] New flows in presenters unit tests
- [ ] Mappers/Validators with any kind of logic unit tests
- [ ] Functional tests pass

### CODE REVIEW

File: app/src/main/java/cm/aptoide/pt/socialMedia/SocialMediaNavigator.kt [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import android.content.Context
+import cm.aptoide.aptoideviews.socialmedia.SocialMediaView
+import cm.aptoide.pt.R
+import cm.aptoide.pt.link.CustomTabsHelper
+import cm.aptoide.pt.themes.ThemeManager
+
+class SocialMediaNavigator(val context: Context, val themeManager: ThemeManager) {
+
+  companion object {
+    var FACEBOOK_URL = "https://pt-pt.facebook.com/aptoide/"
```
Comment (jdandrade) [inline app/src/main/java/cm/aptoide/pt/socialMedia/SocialMediaNavigator.kt:12]: We want this hardcoded for pt? What about other languages, I guess facebook redirects to the right language but still, can we have a generic url?

File: app/src/main/java/cm/aptoide/pt/socialMedia/SocialMediaNavigator.kt [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    var TWITTER_URL = "https://twitter.com/aptoide"
+    var INSTAGRAM_URL = "https://www.instagram.com/aptoideteam/"
+  }
+
+  fun navigateToSocialMediaWebsite(socialMediaType: SocialMediaView.SocialMediaType) {
+    val socialMediaUrl: String = mapSocialMediaTypeToUrl(socialMediaType)
+    CustomTabsHelper.getInstance()
+        .openInChromeCustomTab(socialMediaUrl, context,
+            themeManager.getAttributeForTheme(
+                R.attr.colorPrimary).resourceId)
```
Comment (jdandrade) [inline app/src/main/java/cm/aptoide/pt/socialMedia/SocialMediaNavigator.kt:22]: In my opinion we should have an ExternalNavigator which would unduplicate this logic. We seem to be doing this type of navigation accross the app but there is no standardization.

File: app/src/main/java/cm/aptoide/pt/socialMedia/SocialMediaAnalytics.kt [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+import cm.aptoide.analytics.AnalyticsManager
+import cm.aptoide.analytics.implementation.navigation.NavigationTracker
+import cm.aptoide.aptoideviews.socialmedia.SocialMediaView
+import java.util.*
+
+open class SocialMediaAnalytics(val analyticsManager: AnalyticsManager,
+                                val navigationTracker: NavigationTracker) {
+
+  companion object {
+    @JvmField
```
Comment (jdandrade) [inline app/src/main/java/cm/aptoide/pt/socialMedia/SocialMediaAnalytics.kt:12]: Just curious, are we doing this on purpose? This tells the Kotlin to not  generate getters/setters for this field

**Review by jdandrade — CHANGES_REQUESTED** [review]

File: app/src/main/java/cm/aptoide/pt/socialMedia/SocialMediaAnalytics.kt (hunk 3 above)
Comment (phillipsgoncalves) [reply app/src/main/java/cm/aptoide/pt/socialMedia/SocialMediaAnalytics.kt]: adding this allows to expose it to the outside. I was wronlgy using this. I should add const val.
Also changed the other ones to val as there was no need for them to be a var

**Review by jdandrade — APPROVED** [review]

**Pull request merged**